force-app/main/default/classes/ServiceScheduleService_TEST.cls

```diff
@@ -118,6 +122,173 @@ public with sharing class ServiceScheduleService_TEST {
         );
     }
 
+    @IsTest
+    private static void shouldGetSessions() {
+        String createMethod = 'createRRule';
+        String generateMethod = 'generate';
+        String rRuleString = 'abc';
+        Id primaryServiceProvider = TestUtil.mockId(Contact.SObjectType);
+        Id otherServiceProvider = TestUtil.mockId(Contact.SObjectType);
+        DateTime firstSessionStart = System.now();
+        DateTime firstSessionEnd = System.now().addHours(2);
+        ServiceSchedule__c schedule = new ServiceSchedule__c(
+            Name = 'test',
+            FirstSessionStart__c = firstSessionStart,
+            FirstSessionEnd__c = firstSessionEnd,
+            Interval__c = 1,
+            Frequency__c = 'DAILY',
+            ServiceScheduleEndDate__c = System.today().addDays(10),
+            NumberOfServiceSessions__c = 2,
+            DaysOfWeek__c = '1;2;3;4;5;6;7',
+            PrimaryServiceProvider__c = primaryServiceProvider,
+            OtherServiceProvider__c = otherServiceProvider
+        );
+        List<DateTime> expectedDateTimes = new List<DateTime>{
+            firstSessionStart,
+            firstSessionStart.addDays(1)
+        };
+        recurrenceServiceStub.withReturnValue(
+            createMethod,
+            new List<Type>{
+                String.class,
+                Integer.class,
+                Date.class,
+                Integer.class,
+                List<Integer>.class
+            },
+            rRuleString
+        );
+        recurrenceServiceStub.withReturnValue(
+            generateMethod,
+            new List<Type>{ DateTime.class, String.class },
+            expectedDateTimes
+        );
+        service.recurrenceService = (RecurrenceService) recurrenceServiceStub.createMock();
+
+        List<ServiceSession__c> expected = new List<ServiceSession__c>();
+        for (DateTime sessionDateTime : expectedDateTimes) {
+            expected.add(
+                new ServiceSession__c(
+                    Name = schedule.Name + ': ' + sessionDateTime.date().format(),
+                    SessionStart__c = sessionDateTime,
+                    SessionEnd__c = sessionDateTime.addHours(2),
+                    PrimaryServiceProvider__c = primaryServiceProvider,
+                    OtherServiceProvider__c = otherServiceProvider,
+                    Status__c = 'Pending'
+                )
+            );
+        }
+
+        Test.startTest();
+        List<ServiceSession__c> actual = service.getSessions(schedule);
+        Test.stopTest();
+
+        System.assertEquals(expected, actual);
```

Comment:
@bethbrains , Can we add a message that the users will see if this assert fails.